pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -99,10 +102,30 @@ func (c *EtcdCertSignerController) syncAllMasters(recorder events.Recorder) erro
 
 	errs := []error{}
 	for _, node := range nodes {
-		if err := c.createSecretForNode(node, recorder); err != nil {
+		nodeInternalIPs, err := dnshelpers.GetInternalIPAddressesForNodeName(node)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if err := c.checkSecrets(node, nodeInternalIPs, recorder); err != nil {
+			if errors.IsNotFound(err) {
+				// if any of the secrets is not present, create all secrets for the node
+				if err := c.createSecretsForNode(node, nodeInternalIPs, recorder); err != nil {
```

Comment:
I think this would make sense to just include in `checkSecrets()`? Maybe then we are `ensureSecrets()`?